src/sentry/api/endpoints/organization_dashboard_details.py

```diff
@@ -44,3 +122,40 @@ def delete(self, request, organization, dashboard):
         dashboard.save()
 
         return self.respond(status=204)
+
+    def put(self, request, organization, dashboard):
+        """
+        Edit an Organization's Dashboard
+        ```````````````````````````````````
+
+        Edit an individual organization's dashboard as well as
+        bulk edits on widgets (i.e. rearranging widget order).
+
+        :pparam string organization_slug: the slug of the organization the
+                                          dashboard belongs to.
+        :pparam int dashboard_id: the id of the dashboard.
+        :param array widgets: the array of widgets (consisting of a widget id and the order)
+                            to be updated.
+        :auth: required
+        """
+        serializer = DashboardWithWidgetsSerializer(
+            data=request.DATA,
+            context={'dashboard_id': dashboard.id}
+        )
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        data = serializer.object
+        try:
+            with transaction.atomic():
+                title = data.get('title')
+                if title:
+                    dashboard.update(title=data['title'])
+
+                widgets = data.get('widgets')
+                if widgets:
+                    reorder_widgets(dashboard.id, widgets)
+        except IntegrityError:
+            return self.respond({'Dashboard with that title already exists'}, status=400)
```

Comment:
@markstory is there a better response here? I assume that's probably the most likely thing to have happen, but ... I do so much with widgets it makes me nervous that something could have gone wrong.

Reply:
409 - conflict is a slightly better response code.